src/TLuaInterpreter.cpp

```diff
@@ -2971,6 +2971,35 @@ int TLuaInterpreter::setModulePriority(lua_State* L)
     return 0;
 }
 
+int TLuaInterpreter::getModuleSync(lua_State* L)
+{
+  QString moduleName;
+  if (!lua_isstring(L, 1)) {
+    lua_pushstring(L, "getModuleSync: Module must be a string");
+    return lua_error(L);;
+  } else {
+    moduleName = lua_tostring(L, 1);
+  }
+
+  Host& host = getHostFromLua(L);
+  QMap<QString, QStringList> modules = host.mInstalledModules;
+  if (!modules.contains(moduleName)) {
```

Comment:
It doesn't strike me that the module not being present is a coding (argument type) error - to me it seems to be a run-time error - as such using `lua_error(...)` aborts the called function with no way to tidy up gracefully; it would be better *I think* to do the more friendly "return a nil and an error message" - which is the typical way that Lua uses to say "*You want to do what? Nope, you can't do that with those at this time!*"

Also, the module name may contain non-ASCII characters - so you must converted it back to UTF-8 before you can safely report it back to the user from a `QString`.

Given that `modules` is only used once I would not bother to get a local instance (though the `QMap` type is one that benefits from Qt's [CopyOnWrite](https://en.wikipedia.org/wiki/Copy-on-write#Examples) methodology) and instead access it directly, i.e. go from:
```cpp
    QMap<QString, QStringList> modules = host.mInstalledModules;
    if (!modules.contains(moduleName)) {
	lua_pushfstring(L, "getModuleSync: module %s not found", moduleName);
        return lua_error(L);
    }
```
to:
```cpp
    if (!host.mInstalledModules.contains(moduleName)) {
        lua_pushnil(L);
	lua_pushfstring(L, "module \"%s\" not found", moduleName.toUtf8().constData());
        return 2;
    }
```

Reply:
I just saw this! Thank you for looking! Give me a few. Unfortunately building from local has been a gigantic pain in my rear, which is why I've been leveraging it up here.

Reply:
Yep, making use of Lua's multiple return values. The (`int`) number in the C/C++ `return` statement tells the Lua interpreter how many arguments it is expected to pop off the top of the stack when it returns from the C/C++ function/method. The `return lua_error(L)` is a red-herring in that the `return` ***is never executed*** as `lua_error(L)` does a [**`longjmp()`**](https://en.wikipedia.org/wiki/Setjmp.h) ...